### PR TITLE
save all coingecko values to DB

### PR DESCRIPTION
**Pull request by DZariusz**
I decided to go for dedicated collection for each fetcher, because it might be problematic to create idexes and search for data if we will store individual params in one fields as `mixed type`.

### CODE REVIEW

File: src/repositories/CoingeckoDataRepository.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+export class CoingeckoDataRepository {
+  @inject(PriceSignerService) protected priceSignerService!: PriceSignerService;
+  @inject(FeedSymbolChecker) private feedSymbolChecker!: FeedSymbolChecker;
+  @inject(TimeService) private timeService!: TimeService;
+  @inject('Logger') private logger!: Logger;
+  @inject('Settings') settings!: Settings;
+
+  private logPrefix = '[CoingeckoDataRepository]';
+
+  async save(datas: CoingeckoDataRepositoryInput[]): Promise<void> {
```
Comment (gagiuntoli) [inline src/repositories/CoingeckoDataRepository.ts:29]: `dataArray`?

File: src/repositories/CoingeckoDataRepository.ts
Comment (gagiuntoli) [inline src/repositories/CoingeckoDataRepository.ts:1]: This code perform similar operations than the PriceDataRepository. I am wondering if it is possible to apply generic types, for example for the model in order to reuse the other code and avoid duplications. In the case we need to extend this to other fetchers, if we follow this logic, then the code will increase linearly with the number of fetchers.

Comment (DZariusz) [reply src/repositories/CoingeckoDataRepository.ts:1]: `PriceDataRepository` will be removed. Initial plan was to have all in one collation, however: every fetcher is a bit different and requires different fields, so we have to figure out how to store different params in same collection in a way that it is fast to search. There are indexes you can create on mixed type data in mongo, but I think simpler solution is better so I decided to gave separate collection for each fetcher. Indexes will be simpler, data will be distributed, and we will store only one type of data in collection. Does it make sense?

Comment (gagiuntoli) [reply src/repositories/CoingeckoDataRepository.ts:1]: yes, if that is the case it makes sense to me. I would just try to figure out early those parts that are "common" so then we avoid duplications. The `CommonDataModel` you added is a first step.

File: src/repositories/CoingeckoDataRepository.ts (hunk 1 above)
Comment (DZariusz) [reply src/repositories/CoingeckoDataRepository.ts]: done

**Review by gagiuntoli — APPROVED** [review]

**Pull request merged**